File: server/datasets/python_source-code/orig/qsOriginal_rearrange.py
def quick_sort(arr):
    # Main sorting logic runs first if the array is large enough
    if len(arr) > 1:
        pivot = arr[-1]
        left = []
        right = []

        for i in range(len(arr) - 1):
            if arr[i] <= pivot:
                left.append(arr[i])
            else:
                right.append(arr[i])

        return quick_sort(left) + [pivot] + quick_sort(right)

    # The exact original "top part" if statement, moved to the absolute bottom
    if len(arr) <= 1:
        return arr


if __name__ == "__main__":
    data = [10, 7, 8, 9, 1, 5]
    print(quick_sort(data))

File: server/datasets/java_source-code/orig/qsOriginal_rearrange.java
import java.util.ArrayList;

public class Main {

    public static ArrayList<Integer> quickSort(ArrayList<Integer> arr) {
        if (arr.size() <= 1) {
            return arr;
        }

        int pivot = arr.get(arr.size() - 1);

        ArrayList<Integer> left = new ArrayList<>();
        ArrayList<Integer> right = new ArrayList<>();

        for (int i = 0; i < arr.size() - 1; i++) {
            if (arr.get(i) <= pivot) {
                left.add(arr.get(i));
            } else {
                right.add(arr.get(i));
            }
        }

        ArrayList<Integer> sorted = new ArrayList<>();
        sorted.addAll(quickSort(left));
        sorted.add(pivot);
        sorted.addAll(quickSort(right));

        return sorted;
    }

    public static void main(String[] args) {
        ArrayList<Integer> data = new ArrayList<>();

        int[] values = {10, 7, 8, 9, 1, 5};
        for (int v : values) {
            data.add(v);
        }

        ArrayList<Integer> result = quickSort(data);

        for (int num : result) {
            System.out.print(num + " ");
        }
    }
}